### PR TITLE
Add the "Date Modified" column to preset view

### PR DESCRIPTION
**Pull request by vickyliu-go4it**
Add the "Date Modified" column to the preset view. The time format is the same as the time format in capture file list.

Bug: http://b/187410705
Test: Unit test added.

After the change: https://screen/42q7jBEeoL4RbFz

### CODE REVIEW

File: src/OrbitBase/File.cpp [hunk 1]
```diff
@@ -281,4 +295,20 @@ ErrorMessageOr<std::vector<std::filesystem::path>> ListFilesInDirectory(
   return files;
 }
 
+ErrorMessageOr<std::string> GetFormatedFileDateModified(const std::filesystem::path& path,
+                                                        absl::string_view time_format) {
```
Comment (dimitry-) [inline src/OrbitBase/File.cpp:299]: Maybe return absl::Time here and do the formatting separately. I think we already have the formatTime function somehwere.

File: src/OrbitBase/File.cpp [hunk 2]
```diff
@@ -60,6 +63,17 @@ ssize_t pwrite(int fd, const void* buffer, size_t size, int64_t offset) {
 
 #endif
 
+namespace {
+
+template <typename TimePoint>
+std::time_t to_time_t(TimePoint time_point) {
```
Comment (dimitry-) [inline src/OrbitBase/File.cpp:69]: maybe add a todo to switch this to file_clock::to_sys() once we migrate to C++20.

File: src/DataViews/PresetsDataView.cpp [hunk 3]
```diff
@@ -33,17 +34,25 @@ constexpr const char* kLoadableColumnName = "Loadable";
...
 
 namespace {
 std::string GetLoadStateString(orbit_data_views::AppInterface* app, const PresetFile& preset) {
   orbit_data_views::PresetLoadState load_state = app->GetPresetLoadState(preset);
   return load_state.GetName();
 }
+
+std::string GetDateModifiedString(const PresetFile& preset) {
+  auto datetime_or_error = orbit_base::GetFormatedFileDateModified(preset.file_path());
+  if (datetime_or_error.has_error()) return "";
```
Comment (dimitry-) [inline src/DataViews/PresetsDataView.cpp:53]: also we lose the error message here, maybe add a ERROR() here?

File: src/OrbitBase/File.cpp (hunk 2 above)
Comment (vickyliu-go4it) [reply src/OrbitBase/File.cpp:69]: Done! Thanks for the suggestion!

File: src/DataViews/PresetsDataView.cpp (hunk 3 above)
Comment (vickyliu-go4it) [reply src/DataViews/PresetsDataView.cpp]: Added accordingly. Thanks!

File: src/OrbitBase/File.cpp (hunk 1 above)
Comment (vickyliu-go4it) [reply src/OrbitBase/File.cpp:300]: Thanks for the suggestion!
For the time formatting, we have `FormatTimeWithUnderScores` defined in CaptureSerializer.cpp . It is used in the capture filename and only supports the `%Y_%m_%d_%H_%M_%S` format.

Here I would like to let the time formatting in preset view to be the same as the formatting in the capture file list in the connection window, i.e., `%m/%d/%Y %H:%M %p`.

Comment (dimitry-) [reply src/OrbitBase/File.cpp:300]: Still a good idea to separate formatting from getting modified time. You can add a separate function for formatting time and test it separately.

Comment (vickyliu-go4it) [reply src/OrbitBase/File.cpp]: Make sense! Changed to return `absl::Time` here. Thanks!

**Review by beckerhe — APPROVED** [review]
Thanks Vickly. Nothing on top of what Dimitry already raised. So LGTM!

File: src/DataViews/include/DataViews/PresetsDataView.h [hunk 4]
```diff
@@ -18,7 +18,12 @@
 #include "PresetFile/PresetFile.h"
 #include "preset.pb.h"
 
+namespace orbit_data_views_internal {
+std::string FormatShortDatetime(absl::Time time);
+}  // namespace orbit_data_views_internal
```
Comment (dimitry-) [inline src/DataViews/include/DataViews/PresetsDataView.h:23]: if you want to make it DataView component internal you can move it to a private header. Private headers are header files located in src/DataView. This way they are not accessible for other components but can still be tested.

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: src/DataViews/include/DataViews/PresetsDataView.h (hunk 4 above)
Comment (vickyliu-go4it) [reply src/DataViews/include/DataViews/PresetsDataView.h]: Thanks! I moved this to `DataViewUtils.h` file and added a test for it

**Pull request merged**